docs/versioned-plugins/filters/elasticsearch-v3.19.0.asciidoc

```diff
@@ -0,0 +1,758 @@
+[source, json]
+[
+  {
+    "timestamp": "2025-04-10T12:00:00",
+    "user_id": 123,
+    "action": "login",
+    "status": {
+        "code": 200,
+        "desc": "Success"
+    }
+  },
+  {
+    "timestamp": "2025-04-10T12:05:00",
+    "user_id": 456,
+    "action": "purchase",
+    "status": {
+        "code": 403,
+        "desc": "Forbidden (unauthorized user)"
+    }
+  }
+]
+
+NOTE: If your index has a mapping with sub-objects where `status.code` and `status.desc` actually dotted fields, they appear in {ls} events as a nested structure.
+
+[id="{version}-plugins-{type}s-{plugin}-esql-multifields"]
+===== Conflict on multi-fields
+
+{esql} query fetches all parent and sub-fields fields if your {es} index has https://www.elastic.co/docs/reference/elasticsearch/mapping-reference/multi-fields[multi-fields] or https://www.elastic.co/docs/reference/elasticsearch/mapping-reference/subobjects[subobjects].
+Since {ls} events cannot contain parent field's concrete value and sub-field values together, the plugin ignores sub-fields with warning and includes parent.
+We recommend using the `RENAME` (or `DROP` to avoid warning) keyword in your {esql} query explicitly rename the fields to include sub-fields into the event.
+
+This is a common occurrence if your template or mapping follows the pattern of always indexing strings as "text" (`field`) + " keyword" (`field.keyword`) multi-field.
+In this case it's recommended to do `KEEP field` if the string is identical and there is only one subfield as the engine will optimize and retrieve the keyword, otherwise you can do `KEEP field.keyword | RENAME field.keyword as field`.
+
+To illustrate the situation with example, assuming your mapping has a time `time` field with `time.min` and `time.max` sub-fields as following:
+[source, ruby]
+    "properties": {
+        "time": { "type": "long" },
+        "time.min": { "type": "long" },
+        "time.max": { "type": "long" }
+    }
+
+The {esql} result will contain all three fields but the plugin cannot map them into {ls} event.
+To avoid this, you can use the `RENAME` keyword to rename the `time` parent field to get all three fields with unique fields.
+[source, ruby]
+    ...
+    query => 'FROM my-index | RENAME time AS time.current'
+    ...
+
+For comprehensive ES|QL syntax reference and best practices, see the https://www.elastic.co/guide/en/elasticsearch/reference/current/esql-syntax.html[{esql} documentation].
+
+[id="{version}-plugins-{type}s-{plugin}-options"]
+==== Elasticsearch Filter Configuration Options
+
+This plugin supports the following configuration options plus the <<{version}-plugins-{type}s-{plugin}-common-options>> and the <<{version}-plugins-{type}s-{plugin}-deprecated-options>> described later.
+
+[cols="<,<,<",options="header",]
+|=======================================================================
+|Setting |Input type|Required
+| <<{version}-plugins-{type}s-{plugin}-aggregation_fields>> |<<hash,hash>>|No
```

Comment:
> https://github.com/elastic/logstash-docs/pull/1910#issuecomment-3281851648

This manually generated PR is using the old `<<hash,hash>>` (8x and Asciidoc compatible), but the references should be in this format: `{logstash-ref}/configuration-file-structure.html#hash[hash]` to link correctly in 9.0+.  

Off the top of my head, I'm not sure why this isn't a problem for auto-generated plugin docs. (Check out the PR where [4.2.0](https://github.com/elastic/logstash-docs/pull/1859/files#diff-0c9e5988568af532cdb22e252017f0df653a7bc2ef068fe2effbe6e039cd1e98) was updated.)

<img width="2189" height="1019" alt="Screenshot 2025-09-15 at 7 54 39 PM" src="https://github.com/user-attachments/assets/a6b57290-5841-4b4b-9591-15547a34df67" />

Reply:
it's not just the format that has changed.  The file location has changed, too.

Reply:
Ah didn't notice that we had `--repair` option, will fix this in a moment.